src/assets/scripts/airline/AirlineModel.js

```diff
@@ -273,9 +277,9 @@ export default class AirlineModel {
      */
     _generateAircraft(options) {
         if (!options.callsign) {
-            // TODO: Why is the `AircraftController` repsonsible for generating a callsign, but this is responsible for
-            // generating the flight number? callsign generation logi should live in this class.
-            options.callsign = window.aircraftController.aircraft_callsign_new(options.airline);
+            options.callsign = this.generateFlightNumber();
```

Comment:
@erikquinn `options.callsign` will _always_ have a unique flightNumber.  I've updated the `generateFlightNumber` method to only return a value if it doesn't already in the `callsigns` list.  It's hidden, and it took me a few minutes to find it in the delta:
```
if (window.aircraftController.isCallsignInList(flightNumber)) {
    return this.generateFlightNumber();
}
```

yay recursion!